apps/calendar/js/views/current_time.js

```diff
@@ -60,7 +63,7 @@ CurrentTime.prototype = {
       // inside timespan (eg. current time is 2014-05-22T23:59:50 and user is
       // viewing 2014-05-23 until past midnight)
       this._clearInterval();
-      this._interval = setTimeout(this.activate.bind(this), diff);
+      timeObserver.dayChange.once(this.activate);
```

Comment:
problem with `on('daychange'` is that a typo will go unnoticed.. that's the main reason why I prefer signals.